test/extended/networking/external_gateway.go

```diff
@@ -0,0 +1,60 @@
+package networking
+
+import (
+	"context"
+	"net"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-network] external gateway address", func() {
+	oc := exutil.NewCLI("ns-global")
+
+	InOVNKubernetesContext(func() {
+		f := oc.KubeFramework()
+
+		g.It("Should validate failure if an external gateway address does not match the address family of the pod", func() {
+			_, err := createPod(f.ClientSet, f.Namespace.Name, "test-valid-gateway-pod")
+			expectNoError(err)
+			// Set external gateway address into an IPv6 address and make sure
+			// pod ip address matches with ipv6 address  family.
+			makeNamespaceWithExternalGatewaySet(f, "fd00:10:244:2::6")
+			podIPs, err := createPod(f.ClientSet, f.Namespace.Name, "test-invalid-gateway-pod")
+			if err != nil {
+				o.Expect(len(podIPs)).To(o.Equal(0))
+			} else {
+				o.Expect(len(podIPs)).NotTo(o.Equal(0))
+				for _, podIP := range podIPs {
+					ip := net.ParseIP(podIP.IP)
+					o.Expect(ip.To4()).To(o.BeNil())
+				}
+			}
```

Comment:
It looks like this check always uses an IPv6 address.  Would it be possible to check that an IPv6-only cluster won't get pods with IPv4 external gateways as well? And a dual stack cluster could get either?

Reply:
Thanks @stbenjam, now added checks covering all cases, but I'm not sure about behavior for dual stack, please have a look and let me know if that's okay.